src/AutorestSwift/Code Generator/SwiftGenerator.swift

```diff
@@ -43,6 +43,72 @@ class SwiftGenerator: CodeGenerator {
 
     // MARK: Methods
 
+    fileprivate func generateModels() throws {
+        // Simple dictionary (for fast lookup) to determine if a model has already been written.
+        // Skips regenerating a file with a less specific schema (Schema vs ObjectSchema, for example)
+        // but keeps track of the number of times the model name appears for debugging purposes.
+        var modelsWritten = [String: Int]()
+
+        // Create model files
+        for object in model.schemas.objects ?? [] {
+            let name = object.name
+            guard modelsWritten[name] == nil else {
+                SharedLogger.warn("\(name) has already been generated once this run. Skipping...")
+                modelsWritten[name]! += 1
```

Comment:
how does  modelsWritten[name] int Value being used? From what I can see, it sees like we can use Bool instead of Int in this lookup dictionary

Reply:
That's true, but it doesn't actually matter what we use. We only look to see if the value is nil, meaning there hasn't been an entry. I originally used `Bool` but switched to `Int` since, for debugging purposes, it might be useful to see how many times a file would have otherwise been regenerated. For example, if you see that one file has a much larger count than others, it might point to a name conflict.